pylint/pyreverse/inspector.py

```diff
@@ -178,7 +179,17 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
             for assignattr in assignattrs:
                 if not isinstance(assignattr, nodes.Unknown):
                     self.handle_assignattr_type(assignattr, node)
-        # resolve implemented interface
+        # resolve instance properties
+        for method in node.methods():
+            if decorated_with_property(method) and method.returns is not None:
+                value = list(method.returns.infer())[0]
+                if value.name == "list":
+                    self.handle_property_type(method.name, {method.returns}, node)
+                else:
+                    self.handle_property_type(
+                        method.name, utils.infer_node(value), node
+                    )
+
```

Comment:
I am having some trouble understanding how this addition relates to the scope of this PR. Can you elaborate why this is necessary?